core/src/main/java/org/bitcoinj/script/ScriptBuilder.java

```diff
@@ -351,10 +351,17 @@ public static Script createOpReturnScript(byte[] data) {
      * Create script data bytes to represent the given block height.
      */
     public static byte[] createHeightScriptData(final int height) {
-        // TODO: Replace with something generic to any integer value
         final byte[] int32Buffer = ByteBuffer.allocate(4).order(ByteOrder.LITTLE_ENDIAN).putInt(height).array();
-        if (int32Buffer[3] == 0) {
-            return Arrays.copyOf(int32Buffer, 3);
+
+        // Strip any zeroes from the end
+        int size;
+        for (size = 0; size < int32Buffer.length; size++) {
```

Comment:
You can't do this. To strip zeros you have to go from the back (it's little endian format).